core/match_pattern.py

```diff
@@ -62,44 +62,99 @@ def compute_child_table(suf_tab, lcp_tab):
     return child_table
 
 
-def get_child_intervals(i, j, child_tab: list[Child]) -> list[Tuple[int, int]]:
-    intervals = []
+def is_interval_eligible(a: str, i: int, lcp: int, s: str, suf_tab: list[int]) -> bool:
+    idx_to_hit = suf_tab[i]
+
+    return s[idx_to_hit + lcp] == a
+
+
+def get_lcp(i: int, j: int, child_tab: list[Child], lcp_tab: list[int]):
+    assert i > -1 and j <= len(child_tab)
+
+    # root interval
+    if i == 0 and j == len(child_tab):
+        return 0
+    elif len(child_tab) > j > child_tab[j].up > i:
+        return lcp_tab[child_tab[j].up]
+    else:
+        return lcp_tab[child_tab[i].down]
+
+
+def get_interval(i: int, j: int, a: str, s: str, child_tab: list[Child],
+                 lcp_tab: list[int], suf_tab: list[int]) -> Tuple[int, int]:
```

Comment:
You can use lower-case `tuple[int, int]` just like you use lower-case `list[Child]`.